### PR TITLE
Roll Web Platform Tests

### PR DESCRIPTION
**Pull request by Zirro**
A few changes to the code were made as the `/` prefix has been removed from the paths in the manifest.

### CODE REVIEW

File: test/web-platform-tests/to-run.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ basic.htm: [fail, Unknown]
 client-hint-request-headers-2.tentative.htm: [fail, Client Hints not implemented]
 client-hint-request-headers.htm: [fail-slow, Client Hints not implemented]
 cors-safelisted-request-header.any.html: [fail, Depends on fetch]
-credentials-flag.htm: [fail, Unknown]
+credentials-flag.htm: [needs-node12, Unknown]
```
Comment (domenic) [inline test/web-platform-tests/to-run.yaml]: Weird.

File: test/web-platform-tests/to-run.yaml [hunk 2]
```diff
@@ -126,7 +126,7 @@ DIR: dom/nodes
 Comment-constructor.html: [fail, Unknown]
 DOMImplementation-createDocument.html: [fail, Unknown]
 Document-URL.html: [fail, Unknown]
-Document-characterSet-normalization.html: [fail, Some encodings are not supported - see the whatwg-encoding module]
+Document-characterSet-normalization.html: [timeout, Some encodings are not supported - see the whatwg-encoding module]
```
Comment (domenic) [inline test/web-platform-tests/to-run.yaml]: Why this change?

File: test/web-platform-tests/to-run.yaml (hunk 1 above)
Comment (Zirro) [reply test/web-platform-tests/to-run.yaml]: Yep, presumably a fix from the new HTTP parser.

File: test/web-platform-tests/to-run.yaml (hunk 2 above)
Comment (Zirro) [reply test/web-platform-tests/to-run.yaml]: With v12, I saw this test take various times between 10000ms to 60000ms until it failed.

Comment (Zirro) [reply test/web-platform-tests/to-run.yaml]: Interestingly it only seems to occur when all tests up to that point have been run. If just that test or only the tests from that directory run in isolation, I don't see the same long delay.

**Comment by Zirro** [comment]
Looks like there needs to be a new release of canvas with Node.js v12 support: https://github.com/Automattic/node-canvas/issues/1410

I'll disable testing of the canvas dependency in stable Node for now.

**Comment by Zirro** [comment]
Rolled again to include the tests for https://github.com/jsdom/jsdom/pull/2571.

I also ended up disabling `response-method.htm` as `flaky`, since I saw it passing a couple of times even though it still fails most of the time in Node.js v12.
